lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -258,48 +258,61 @@ namespace Js
         {
             pattern = scriptContext->GetLibrary()->GetEmptyRegexPattern();
         }
-        else if (JavascriptRegExp::Is(args[1]))
+        else if (JavascriptRegExp::IsRegExpLike(args[1], scriptContext))
```

Comment:
will you run the benchmark to verify the perf? (IsRegExpLike seems to costly at this point).